rust/src/lib.rs

```diff
@@ -1864,13 +1803,252 @@ pub unsafe extern "C" fn zcashlc_rewind_to_height(
     unwrap_exc_or(res, false)
 }
 
+/// A struct that contains a subtree root.
+///
+/// # Safety
+///
+/// - `root_hash_ptr` must be non-null and must be valid for reads for `root_hash_ptr_len`
+///   bytes, and it must have an alignment of `1`.
+/// - The total size `root_hash_ptr_len` of the slice pointed to by `root_hash_ptr` must
+///   be no larger than `isize::MAX`. See the safety documentation of `pointer::offset`.
+#[repr(C)]
+pub struct FfiSubtreeRoot {
+    root_hash_ptr: *mut u8,
+    root_hash_ptr_len: usize,
```

Comment:
Why a pointer and a length here? These should all be pointers to 32 byte arrays, right?

Reply:
I copied this FFI pattern from `FFIBlockMeta` where it is used for block hashes. But it is indeed currently the case that all root hashes we'd see here are 32 bytes. I'll leave this one as-is since it's been tested and changing it would require more changes to the Kit, but I'll add subsequent similar fields without the length.